### ISSUE TITLE
int arrays are empty while floats work

### ISSUE THREAD

**Issue by enchantedBunny** [wouterboomsma/eigency#20]
Hey, thanks a lot for Eigency, it's exactly what I need.

While playing around with it and getting it work, I was really confused by the fact that normal integers become zero when passed to C++.

I made a little function (copy of the one from readme) which outputs the size, cols and rows as well as the matrix element at a time.

![image](https://user-images.githubusercontent.com/13186803/32168290-3e2571d6-bd6c-11e7-9345-775aec239597.png)

As you can see here, it works great for float:

![image](https://user-images.githubusercontent.com/13186803/32168237-0ddc7d1c-bd6c-11e7-80c7-6e56155bdbaf.png)

Is this intentional? What am I doing wrong?

Also, I'm using std::to_string while printing it through a python function, I don't think that could be the problem.

Thanks

**Comment by wouterboomsma** [comment]
I'm very sorry for the very long response time on this. I've been wanting to change some things in eigency to make it more difficult to get into situations like the one you describe, but I simply haven't been able to find the time before today. The issue you describe is expected behavior, because you force the interface to interpret your python integers as c++ float values. I've now added a section on this in the README file ("Types matter"), and I've now changed all examples (and tests) to use explicitly types numpy arrays in the definitions of the Cython interface, which avoids the surprising behaviour you experienced (also explained in the "Types matter" section).
